File: Experiment Processing/recommender/voting_rules/most_pleasure.py
from recommender.voting_rules.abstract_voting_rule import AbstractVotingRule


class MostPleasure(AbstractVotingRule):
    """
    "Most Pleasure" aggregation strategy. Basically the opposite of Least Misery: returns that maximum rating of each track.
    """

    def get_name(self):
        return "Most Pleasure"

    def voting_rule(self, song, song_ratings):
        return max(song_ratings.values())

File: Web Survey/resources/js/recommender/voting_rules/MostPleasure.js
import AbstractVotingRule from "./AbstractVotingRule";

export default class MostPleasure extends AbstractVotingRule {

    static getRuleName() {
        return "Most Pleasure";
    }

    getDescription() {
        return "First finds the highest rating for every track, then picks the best of these."
    }

    votingRule(song, songRatings) {
        //return max(song_ratings.values())
        return Math.max(...Object.values(songRatings));
    }

}